realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -103,9 +103,11 @@ public boolean generate() {
         }
 
         TypeElement parentElement = (TypeElement) Utils.getSuperClass(classType);
-        if (!parentElement.toString().endsWith(".RealmObject")) {
-            Utils.error("A RealmClass annotated object must be derived from RealmObject", classType);
-            return false;
+        // The only allowed derived class is RealmObject, POJO implementing RealmModel
```

Comment:
This sounds a bit strange to me, perhaps "Realm model classes must either extend RealmObject or implement RealmModel to be considered a valid model class.